[17.01] add release notes

martenson opened the pull request:
Please help with curating these @galaxyproject/core - this is our work being presented. 

**Every committer has push access to this branch on my fork.**

martenson committed:
add initial release notes

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+.. feature
+
+* New User Preferences User Interface
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
```
`@anuprulez <https://github.com/anuprulez>`__ and

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3118`_
+* Per user total walltime limit.
+  (thanks to `@interactomix <https://github.com/interactomix>`__).
```
s/interactomix/maxf130/g

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  `Pull Request 3217`_
+* Add API to install resolver tool dependencies
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3222`_
```
This was backported to 16.10 before its release and already announced in the corresponding release notes, I'd remove it or move it to the `enhancement` section.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3222`_
+* Replace CasperJS browser tests with Selenium tests.
+  `Pull Request 3239`_
```
Not sure this is of enough general interest to be a feature.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  `Pull Request 3106`_
+* Support for gzipped fastq formats
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3145`_
```
This should be moved to `feature` IMO.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3145`_
+* Version 17.01.dev
+  `Pull Request 3192`_
```
Remove, it's minor (I tagged it as such now).

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  `Pull Request 3197`_
+* Fix user table in admin view
+  (thanks to `@erasche <https://github.com/erasche>`__).
+  `Pull Request 3200`_
```
I'd move this to `bug` section.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+* Further tighten up GG format.
+  `Pull Request 3233`_
+* change conda channel order
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
```
s/galaxyproject/bgruening/g

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3301`_
+* Bring in galaxyproject/galaxy-lib`#36
+  <https://github.com/galaxyproject/galaxy/issues/36>`__ from @bgruening.
```
s|galaxy/issues/36|galaxy-lib/pull/36|

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  `Pull Request 3348`_
+* Fuzzy search for tools search
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3356`_
```
s/bgruening/anuprulez/g

Also, move to feature?

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+* Workflow run display improvements - esp. for inputs.
+  `Pull Request 3369`_
+* Expand selenium test coverage for workflows.
+  `Pull Request 3371`_
```
What about collecting all CasperJS and Selenium test entries in a single one with all the PR links?

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  `Pull Request 3381`_
+* Fix typo
+  (thanks to `@pvanheus <https://github.com/pvanheus>`__).
+  `Pull Request 3384`_
```
Move to `bug` section.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+* Resolve Conda Dependencies All at Once
+  `Pull Request 3391`_
+* Markdown datatype support.
+  `Pull Request 3399`_
```
Remove, this was reverted in https://github.com/galaxyproject/galaxy/pull/3408

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  `Pull Request 3405`_
+* Revert "Markdown datatype support."
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
+  `Pull Request 3408`_
```
Remove together with the note about #3399.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  (thanks to `@manabuishii <https://github.com/manabuishii>`__).
+  `Pull Request 3411`_
+* Target Miniconda3 when installing Conda.
+  `Pull Request 3419`_
```
Move to `feature`?

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+* Fix explicit Docker container resolution.
+  `Pull Request 3346`_
+* Masthead Webhook bug fix
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
```
s/bgruening/anuprulez/g

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,417 @@
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
+  `Pull Request 3378`_
+* Add diamond database as datatype
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
```
and `@bebatut <https://github.com/bebatut>`__

nsoranzo reviewed (commented):
Thanks @martenson! I've added some corrections and suggestions.

martenson commented:
@nsoranzo Can you please directly modify and commit your changes? You can do so on GitHub directly:
https://github.com/martenson/galaxy/blob/17.01_release_notes/doc/source/releases/17.01.rst

I believe nobody will dispute your corrections or suggestions on release notes.

edit: I will happily do so myself if it is inconvenient for you or you don't have time, just tell me please.

martenson committed:
bring in @nsoranzo's comments

nsoranzo commented:
@martenson Thanks for doing it and sorry, I didn't have time.

martenson committed:
more work towards reasonable release notes

martenson commented on doc/source/releases/17.01_announce.rst:
```diff
@@ -8,14 +8,21 @@ January 2017 Galaxy Release (v 17.01)
+  Implemented in `Pull Request 3118`_.
+
+**Support for gzipped fastq formats**
+  You can now upload compressed files without being unpacked. Existing tools will work as before and future
```
need to specify which formats.

nsoranzo replied:
s/compressed files/compressed (with gzip or bzip2) FASTQ files/

nsoranzo commented on doc/source/releases/17.01_announce.rst:
```diff
@@ -8,14 +8,21 @@ January 2017 Galaxy Release (v 17.01)
+  Thanks to `@anuprulez <https://github.com/anuprulez>`__ and `@bgruening <https://github.com/bgruening>`__.
+  Implemented in `Pull Request 3118`_.
+
+**Support for gzipped fastq formats**
```
s/gzipped fastq/compressed FASTQ/

nsoranzo reviewed (commented):
Thanks @martenson, brilliant!

martenson committed:
clarify fastq compression handling

martenson commented:
Thanks for your help @nsoranzo !

This is now ready for review @galaxyproject/core

nsoranzo commented on doc/source/releases/17.01_announce.rst:
```diff
@@ -10,17 +10,20 @@ Highlights
-  You can now upload compressed files without being unpacked. Existing tools will work as before and future
-  tools will be able to consume archives and save space in your quota.
+**Support for compressed FASTQ formats**
+  You can now upload compressed (with gzip or bzip2) FASTQ files without them being unpacked automatically (as it was until now).
```
Should we also specify that this applies only if you explicitly assign the corresponding datatype during file upload (sniffing disabled by default)?

martenson replied:
done

martenson committed:
clarify how to prevent unpacking of archives

jmchilton committed:
Don't encourage uploads of fastq in release notes.

My understanding is nearly all data now produced is fastqsanger.

mvdbeek commented on doc/source/releases/17.01_announce.rst:
```diff
@@ -3,8 +3,50 @@
+
+**Support for compressed FASTQ formats**
+  You can now upload compressed (with gzip or bzip2) FASTQ files without them being unpacked automatically (as it was until now).
+  For this to work you have to select the proper type when uploading (e.g. instead of ``fastqsanger`` you select ``fastqsanger.gz`` for compressed files).
```
s/``fastqsanger.gz``/``fastqsanger.gz or fastqsanger.bz2``/

martenson replied:
This is an example, I do not think we need to list more possibilities.

mvdbeek commented on doc/source/releases/17.01_announce.rst:
```diff
@@ -3,8 +3,50 @@
+  You can now upload compressed (with gzip or bzip2) FASTQ files without them being unpacked automatically (as it was until now).
+  For this to work you have to select the proper type when uploading (e.g. instead of ``fastqsanger`` you select ``fastqsanger.gz`` for compressed files).
+  Existing tools will work as before and future tools will be able to consume archives and save space in your quota.
+  Thanks to `@mvdbeek <https://github.com/mvdbeek>`__.
```
`@pvanheus @abretaud @ashok` also contributed commits in that PR,
and I think we should also mention https://github.com/galaxyproject/galaxy/pull/3514 and https://github.com/galaxyproject/galaxy/pull/3510

mvdbeek committed:
Add all PRs and authors for compressed fastq support

martenson committed:
remove jmchilton from credits

the Galaxy-lab members are explicitly omitted from RN mentions

mvdbeek committed:
Add ashvark to author list

mvdbeek commented:
Thanks @martenson, I didn't know we exclude galaxy-lab members.

martenson commented:
@mvdbeek Yeah, if we didn't it would have been too much text I believe. We tend to do some things worthy of release notes too. 😋 

Also we want to highlight everybody else.

jmchilton committed:
Various improvements to the release notes.

- Rewording things to focus on reader concerns.
- Grouping things into single bullets (like testing stuff) and close to similar topics (like Conda stuff).
- Eliminated some things which should have been marked #minor

jmchilton commented:
@martenson As promised a did a major pass at the release notes. Feel free to back out of anything you disagree with and we can discuss it more. I may make another pass as well but this first one was fairly significant.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -9,163 +9,124 @@
+* Fix Conda channel order
+  (thanks to `@bgruening <https://github.com/bgruening>`__).
+  `Pull Request 3237`_
+* Target Miniconda3 when initialized new Conda installs.
```
`when initialized` doesn't sound right to me.

martenson commented:
splendid, thanks @jmchilton

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -9,163 +9,124 @@
+  `Pull Request 3482`_, `Pull Request 3185`_
+* Fix to use lexists instead exists in dependency resolver code
   (thanks to `@bgruening <https://github.com/bgruening>`__).
   `Pull Request 3301`_
```
Move this to the "dependency resolver" section above?

jmchilton replied:
I actually moved this into the bug fixes section. I think that it is fine down there. We could list it under "misc" fixes for dependency resolvers also - do you feel strongly about this?

nsoranzo replied:
That's fine, thanks!

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -9,163 +9,124 @@
-* Update datatype converter tools
+* Large cleanup of datatype converter tools
   (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
   `Pull Request 3305`_
```
Move this to the "datatypes" section above?

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -9,163 +9,124 @@
-* Addition of IGV display for genome fasta file
+* Implement IGV display for FASTA datatypes
   (thanks to `@Delphine-L <https://github.com/Delphine-L>`__).
   `Pull Request 3327`_
```
Move this to the "datatypes" section above?

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -9,163 +9,124 @@
   `Pull Request 3328`_
 * Allow admins to see API keys under REMOTE_USER
   (thanks to `@erasche <https://github.com/erasche>`__).
   `Pull Request 3338`_
```
Maybe we can also group the API PRs.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -9,163 +9,124 @@
+  (thanks to `@abretaud <https://github.com/abretaud>`__).
+  `Pull Request 3348`_
+* Use a newer conda version by default
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
```
s/galaxyproject/bgruening/g

jmchilton replied:
Caught this myself on the second pass.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -174,19 +135,11 @@ Fixes
-  `Pull Request 3370`_
 * Fix dependency display and api
   (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
   `Pull Request 3427`_
```
Move this to the "dependency resolver" section above?

jmchilton replied:
Sorry - I was working on this as you were commenting. I merged this in with the generic fixes just now I think.

nsoranzo reviewed (commented):
Thanks @jmchilton, great work!

jmchilton committed:
More work on 17.01 release notes.

martenson committed:
minor tweaks of RN

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,331 @@
+* Add API to install resolver tool dependencies
+  (thanks to `@mvdbeek <https://github.com/mvdbeek>`__).
+  `Pull Request 3222`_
+* Various other dependency resolver fixes and improvements.
```
Marius should be thanked as author of 2 of the following PRs.

nsoranzo commented on doc/source/releases/17.01.rst:
```diff
@@ -0,0 +1,331 @@
+* Dependency resolver documentation fixes
+  (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
+  `Pull Request 3490`_
+* Fix directory extistence checks in dependency resolver code
```
s/extistence/existence/

martenson committed:
minor changes thanks to @nsoranzo

jmchilton commented:
Thanks for the [cat herding](https://www.fallon.com/case/cat-herders-2/) @martenson!
